src/riak_kv_qry_worker.erl

```diff
@@ -262,7 +262,31 @@ throttling_spawn_index_fsms(State) ->
 estimate_query_size(#state{n_subqueries_done = NSubqueriesDone} = State)
   when NSubqueriesDone < 2  ->
     State;
-
```

Comment:
Yes, checks are performed on every chunk, starting from chunk 2 all the way to the last.

(For the last chunk to cause the query to be cancelled, (1) it needs to be a very large one tipping the average chunk size over the value that was (2) just below `MaxQuerySize / length(SubQrys)` all along.)